spark.yaml

```diff
@@ -0,0 +1,46 @@
+package:
+  name: spark
+  version: 3.5.0
+  epoch: 0
+  description:
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - openjdk-17-jre
+      - bash
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - curl
+      - maven
+      - openjdk-17
+      - openjdk-17-default-jvm
+      - bash
+      - wolfi-base
+      - wolfi-baselayout
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/spark
+      tag: v${{package.version}}
+      expected-commit: ce5ddad990373636e94071e7cef2f31021add07b
+
+  - runs: |
+      export LANG=en_US.UTF-8
+      export JAVA_HOME=/usr/lib/jvm/java-17-openjdk
+
+      ./build/mvn -DskipTests clean install
+      mkdir -p ${{targets.destdir}}/usr/share/java/spark
+      mv /root/.m2/repository/org/apache/spark/* ${{targets.destdir}}/usr/share/java/spark
```

Comment:
I wonder if this copies more that required?  I.e. what ends up in the resulting apk as we may only want a single jar or war?

Reply:
I'm building this locally to see

Reply:
just built this inside a wolfi-base container and running tree in the folder that's being copied into the destdir above would mean these files being in the end apk
```
/root/.m2/repository/org/apache/spark/
|-- spark-avro_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-avro_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-avro_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-avro_2.12-3.5.0-javadoc.jar
|   |   |-- spark-avro_2.12-3.5.0-sources.jar
|   |   |-- spark-avro_2.12-3.5.0-test-sources.jar
|   |   |-- spark-avro_2.12-3.5.0-tests.jar
|   |   |-- spark-avro_2.12-3.5.0.jar
|   |   `-- spark-avro_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-catalyst_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-catalyst_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-catalyst_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-catalyst_2.12-3.5.0-javadoc.jar
|   |   |-- spark-catalyst_2.12-3.5.0-sources.jar
|   |   |-- spark-catalyst_2.12-3.5.0-test-sources.jar
|   |   |-- spark-catalyst_2.12-3.5.0-tests.jar
|   |   |-- spark-catalyst_2.12-3.5.0.jar
|   |   `-- spark-catalyst_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-common-utils_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-common-utils_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-common-utils_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-common-utils_2.12-3.5.0-javadoc.jar
|   |   |-- spark-common-utils_2.12-3.5.0-sources.jar
|   |   |-- spark-common-utils_2.12-3.5.0-test-sources.jar
|   |   |-- spark-common-utils_2.12-3.5.0-tests.jar
|   |   |-- spark-common-utils_2.12-3.5.0.jar
|   |   `-- spark-common-utils_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-connect-client-jvm_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-connect-client-jvm_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-connect-client-jvm_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-connect-client-jvm_2.12-3.5.0-javadoc.jar
|   |   |-- spark-connect-client-jvm_2.12-3.5.0-sources.jar
|   |   |-- spark-connect-client-jvm_2.12-3.5.0-test-sources.jar
|   |   |-- spark-connect-client-jvm_2.12-3.5.0-tests.jar
|   |   |-- spark-connect-client-jvm_2.12-3.5.0.jar
|   |   `-- spark-connect-client-jvm_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-connect-common_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-connect-common_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-connect-common_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-connect-common_2.12-3.5.0-javadoc.jar
|   |   |-- spark-connect-common_2.12-3.5.0-sources.jar
|   |   |-- spark-connect-common_2.12-3.5.0-test-sources.jar
|   |   |-- spark-connect-common_2.12-3.5.0-tests.jar
|   |   |-- spark-connect-common_2.12-3.5.0.jar
|   |   `-- spark-connect-common_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-connect_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-connect_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-connect_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-connect_2.12-3.5.0-javadoc.jar
|   |   |-- spark-connect_2.12-3.5.0-sources.jar
|   |   |-- spark-connect_2.12-3.5.0-test-sources.jar
|   |   |-- spark-connect_2.12-3.5.0-tests.jar
|   |   |-- spark-connect_2.12-3.5.0.jar
|   |   `-- spark-connect_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-core_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-core_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-core_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-core_2.12-3.5.0-javadoc.jar
|   |   |-- spark-core_2.12-3.5.0-sources.jar
|   |   |-- spark-core_2.12-3.5.0-test-sources.jar
|   |   |-- spark-core_2.12-3.5.0-tests.jar
|   |   |-- spark-core_2.12-3.5.0.jar
|   |   `-- spark-core_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-graphx_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-graphx_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-graphx_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-graphx_2.12-3.5.0-javadoc.jar
|   |   |-- spark-graphx_2.12-3.5.0-sources.jar
|   |   |-- spark-graphx_2.12-3.5.0-test-sources.jar
|   |   |-- spark-graphx_2.12-3.5.0-tests.jar
|   |   |-- spark-graphx_2.12-3.5.0.jar
|   |   `-- spark-graphx_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-hive_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-hive_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-hive_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-hive_2.12-3.5.0-javadoc.jar
|   |   |-- spark-hive_2.12-3.5.0-sources.jar
|   |   |-- spark-hive_2.12-3.5.0-test-sources.jar
|   |   |-- spark-hive_2.12-3.5.0-tests.jar
|   |   |-- spark-hive_2.12-3.5.0.jar
|   |   `-- spark-hive_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-kvstore_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-kvstore_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-kvstore_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-kvstore_2.12-3.5.0-javadoc.jar
|   |   |-- spark-kvstore_2.12-3.5.0-sources.jar
|   |   |-- spark-kvstore_2.12-3.5.0-test-sources.jar
|   |   |-- spark-kvstore_2.12-3.5.0-tests.jar
|   |   |-- spark-kvstore_2.12-3.5.0.jar
|   |   `-- spark-kvstore_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-launcher_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-launcher_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-launcher_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-launcher_2.12-3.5.0-javadoc.jar
|   |   |-- spark-launcher_2.12-3.5.0-sources.jar
|   |   |-- spark-launcher_2.12-3.5.0-test-sources.jar
|   |   |-- spark-launcher_2.12-3.5.0-tests.jar
|   |   |-- spark-launcher_2.12-3.5.0.jar
|   |   `-- spark-launcher_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-mllib-local_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-mllib-local_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-mllib-local_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-mllib-local_2.12-3.5.0-javadoc.jar
|   |   |-- spark-mllib-local_2.12-3.5.0-sources.jar
|   |   |-- spark-mllib-local_2.12-3.5.0-test-sources.jar
|   |   |-- spark-mllib-local_2.12-3.5.0-tests.jar
|   |   |-- spark-mllib-local_2.12-3.5.0.jar
|   |   `-- spark-mllib-local_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-mllib_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-mllib_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-mllib_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-mllib_2.12-3.5.0-javadoc.jar
|   |   |-- spark-mllib_2.12-3.5.0-sources.jar
|   |   |-- spark-mllib_2.12-3.5.0-test-sources.jar
|   |   |-- spark-mllib_2.12-3.5.0-tests.jar
|   |   |-- spark-mllib_2.12-3.5.0.jar
|   |   `-- spark-mllib_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-network-common_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-network-common_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-network-common_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-network-common_2.12-3.5.0-javadoc.jar
|   |   |-- spark-network-common_2.12-3.5.0-sources.jar
|   |   |-- spark-network-common_2.12-3.5.0-test-sources.jar
|   |   |-- spark-network-common_2.12-3.5.0-tests.jar
|   |   |-- spark-network-common_2.12-3.5.0.jar
|   |   `-- spark-network-common_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-network-shuffle_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-network-shuffle_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-network-shuffle_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-network-shuffle_2.12-3.5.0-javadoc.jar
|   |   |-- spark-network-shuffle_2.12-3.5.0-sources.jar
|   |   |-- spark-network-shuffle_2.12-3.5.0-test-sources.jar
|   |   |-- spark-network-shuffle_2.12-3.5.0-tests.jar
|   |   |-- spark-network-shuffle_2.12-3.5.0.jar
|   |   `-- spark-network-shuffle_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-parent_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-parent_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-parent_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-parent_2.12-3.5.0-tests.jar
|   |   `-- spark-parent_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-protobuf_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-protobuf_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-protobuf_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-protobuf_2.12-3.5.0-javadoc.jar
|   |   |-- spark-protobuf_2.12-3.5.0-sources.jar
|   |   |-- spark-protobuf_2.12-3.5.0-test-sources.jar
|   |   |-- spark-protobuf_2.12-3.5.0-tests.jar
|   |   |-- spark-protobuf_2.12-3.5.0.jar
|   |   `-- spark-protobuf_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-repl_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-repl_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-repl_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-repl_2.12-3.5.0-javadoc.jar
|   |   |-- spark-repl_2.12-3.5.0-sources.jar
|   |   |-- spark-repl_2.12-3.5.0-test-sources.jar
|   |   |-- spark-repl_2.12-3.5.0-tests.jar
|   |   |-- spark-repl_2.12-3.5.0.jar
|   |   `-- spark-repl_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-sketch_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-sketch_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-sketch_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-sketch_2.12-3.5.0-javadoc.jar
|   |   |-- spark-sketch_2.12-3.5.0-sources.jar
|   |   |-- spark-sketch_2.12-3.5.0-test-sources.jar
|   |   |-- spark-sketch_2.12-3.5.0-tests.jar
|   |   |-- spark-sketch_2.12-3.5.0.jar
|   |   `-- spark-sketch_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-sql-api_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-sql-api_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-sql-api_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-sql-api_2.12-3.5.0-javadoc.jar
|   |   |-- spark-sql-api_2.12-3.5.0-sources.jar
|   |   |-- spark-sql-api_2.12-3.5.0-test-sources.jar
|   |   |-- spark-sql-api_2.12-3.5.0-tests.jar
|   |   |-- spark-sql-api_2.12-3.5.0.jar
|   |   `-- spark-sql-api_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-sql-kafka-0-10_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0-javadoc.jar
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0-sources.jar
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0-test-sources.jar
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0-tests.jar
|   |   |-- spark-sql-kafka-0-10_2.12-3.5.0.jar
|   |   `-- spark-sql-kafka-0-10_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-sql_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-sql_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-sql_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-sql_2.12-3.5.0-javadoc.jar
|   |   |-- spark-sql_2.12-3.5.0-sources.jar
|   |   |-- spark-sql_2.12-3.5.0-test-sources.jar
|   |   |-- spark-sql_2.12-3.5.0-tests.jar
|   |   |-- spark-sql_2.12-3.5.0.jar
|   |   `-- spark-sql_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-streaming-kafka-0-10-assembly_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0-sources.jar
|   |   |-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0-test-sources.jar
|   |   |-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0-tests.jar
|   |   |-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0.jar
|   |   `-- spark-streaming-kafka-0-10-assembly_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-streaming-kafka-0-10_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0-javadoc.jar
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0-sources.jar
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0-test-sources.jar
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0-tests.jar
|   |   |-- spark-streaming-kafka-0-10_2.12-3.5.0.jar
|   |   `-- spark-streaming-kafka-0-10_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-streaming_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-streaming_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-streaming_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-streaming_2.12-3.5.0-javadoc.jar
|   |   |-- spark-streaming_2.12-3.5.0-sources.jar
|   |   |-- spark-streaming_2.12-3.5.0-test-sources.jar
|   |   |-- spark-streaming_2.12-3.5.0-tests.jar
|   |   |-- spark-streaming_2.12-3.5.0.jar
|   |   `-- spark-streaming_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-tags_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-tags_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-tags_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-tags_2.12-3.5.0-javadoc.jar
|   |   |-- spark-tags_2.12-3.5.0-sources.jar
|   |   |-- spark-tags_2.12-3.5.0-test-sources.jar
|   |   |-- spark-tags_2.12-3.5.0-tests.jar
|   |   |-- spark-tags_2.12-3.5.0.jar
|   |   `-- spark-tags_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
|-- spark-token-provider-kafka-0-10_2.12
|   |-- 3.5.0
|   |   |-- _remote.repositories
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0-cyclonedx.json
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0-cyclonedx.xml
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0-javadoc.jar
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0-sources.jar
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0-test-sources.jar
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0-tests.jar
|   |   |-- spark-token-provider-kafka-0-10_2.12-3.5.0.jar
|   |   `-- spark-token-provider-kafka-0-10_2.12-3.5.0.pom
|   `-- maven-metadata-local.xml
`-- spark-unsafe_2.12
    |-- 3.5.0
    |   |-- _remote.repositories
    |   |-- spark-unsafe_2.12-3.5.0-cyclonedx.json
    |   |-- spark-unsafe_2.12-3.5.0-cyclonedx.xml
    |   |-- spark-unsafe_2.12-3.5.0-javadoc.jar
    |   |-- spark-unsafe_2.12-3.5.0-sources.jar
    |   |-- spark-unsafe_2.12-3.5.0-test-sources.jar
    |   |-- spark-unsafe_2.12-3.5.0-tests.jar
    |   |-- spark-unsafe_2.12-3.5.0.jar
    |   `-- spark-unsafe_2.12-3.5.0.pom
    `-- maven-metadata-local.xml
```

Reply:
looks like a lot of test related files + not sure what is expected to be able to run spark?

Reply:
Interesting, makes sense to only copy the required jars over, instead of all the docs, tests etc.

Here's maybe a list from their docker image?

```
docker run --rm -it  apache/spark-py:latest find /opt/spark -name '*spark*.jar'                                          
. . .
+ SPARK_CLASSPATH='/opt/spark/conf::/opt/spark/jars/*'
. . .
/opt/spark/examples/jars/spark-examples_2.12-3.4.0.jar
/opt/spark/jars/spark-tags_2.12-3.4.0.jar
/opt/spark/jars/spark-mesos_2.12-3.4.0.jar
/opt/spark/jars/spark-hive-thriftserver_2.12-3.4.0.jar
/opt/spark/jars/spark-catalyst_2.12-3.4.0.jar
/opt/spark/jars/spark-hive_2.12-3.4.0.jar
/opt/spark/jars/spark-core_2.12-3.4.0.jar
/opt/spark/jars/spark-sketch_2.12-3.4.0.jar
/opt/spark/jars/spark-sql_2.12-3.4.0.jar
/opt/spark/jars/spark-unsafe_2.12-3.4.0.jar
/opt/spark/jars/spark-streaming_2.12-3.4.0.jar
/opt/spark/jars/spark-yarn_2.12-3.4.0.jar
/opt/spark/jars/spark-network-shuffle_2.12-3.4.0.jar
/opt/spark/jars/spark-kvstore_2.12-3.4.0.jar
/opt/spark/jars/spark-network-common_2.12-3.4.0.jar
/opt/spark/jars/spark-mllib-local_2.12-3.4.0.jar
/opt/spark/jars/spark-kubernetes_2.12-3.4.0.jar
/opt/spark/jars/spark-launcher_2.12-3.4.0.jar
/opt/spark/jars/spark-graphx_2.12-3.4.0.jar
/opt/spark/jars/spark-mllib_2.12-3.4.0.jar
/opt/spark/jars/spark-repl_2.12-3.4.0.jar
```

Reply:
So basically we need to do something like `find /opt/spark -name '*spark*.jar` and then have only those jars copied? Will we need setting this also? `SPARK_CLASSPATH='/opt/spark/conf::/opt/spark/jars/*'`

Reply:
yeah that sounds about right to me, might be worth hacking a local spark image to test with

Reply:
Just attempted to fix it with the latest commit, thanks @jamonation for the suggestions and thanks @rawlingsj for the review and help! 
here are the needed jars that we have:
```
[sdk] ❯ cd usr/share/java
[sdk] ❯ cd spark
[sdk] ❯ ls
spark-avro_2.12-3.5.0.jar                           spark-network-shuffle_2.12-3.5.0.jar
spark-catalyst_2.12-3.5.0.jar                       spark-protobuf_2.12-3.5.0.jar
spark-common-utils_2.12-3.5.0.jar                   spark-repl_2.12-3.5.0.jar
spark-connect-client-jvm_2.12-3.5.0.jar             spark-sketch_2.12-3.5.0.jar
spark-connect-common_2.12-3.5.0.jar                 spark-sql-api_2.12-3.5.0.jar
spark-connect_2.12-3.5.0.jar                        spark-sql-kafka-0-10_2.12-3.5.0.jar
spark-core_2.12-3.5.0.jar                           spark-sql_2.12-3.5.0.jar
spark-graphx_2.12-3.5.0.jar                         spark-streaming-kafka-0-10-assembly_2.12-3.5.0.jar
spark-hive_2.12-3.5.0.jar                           spark-streaming-kafka-0-10_2.12-3.5.0.jar
spark-kvstore_2.12-3.5.0.jar                        spark-streaming_2.12-3.5.0.jar
spark-launcher_2.12-3.5.0.jar                       spark-tags_2.12-3.5.0.jar
spark-mllib-local_2.12-3.5.0.jar                    spark-token-provider-kafka-0-10_2.12-3.5.0.jar
spark-mllib_2.12-3.5.0.jar                          spark-unsafe_2.12-3.5.0.jar
```